subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/AbstractIsolatedProjectsIntegrationTest.groovy

```diff
@@ -16,26 +16,19 @@
 
 package org.gradle.configurationcache.isolated
 
-import org.gradle.configurationcache.fixtures.AbstractOptInFeatureIntegrationTest
+import org.gradle.configurationcache.fixtures.AbstractConfigurationCacheOptInFeatureIntegrationTest
 
 import static org.gradle.initialization.StartParameterBuildOptions.IsolatedProjectsOption.PROPERTY_NAME
 
-abstract class AbstractIsolatedProjectsIntegrationTest extends AbstractOptInFeatureIntegrationTest {
+abstract class AbstractIsolatedProjectsIntegrationTest extends AbstractConfigurationCacheOptInFeatureIntegrationTest {
     public static final String ENABLE_CLI = "-D${PROPERTY_NAME}=true"
     final def fixture = new IsolatedProjectsFixture(this)
 
-    @Override
-    void configurationCacheRun(String... tasks) {
+    void isolatedProjectsRun(String... tasks) {
```

Comment:
💭I wonder if some idiom for "running the build with the default settings" would still make some sense across these and other tests... It's usually easy to guess how to run a build by looking at the other tests in the class, though.

Reply:
I would trade the explicit method for code re-use in this case.

We already have a number of generic methods for running builds in `AbstractIntegrationSpec`. I wouldn't want to mix in a couple other generic ones that come from an AbstractXxxIntegrationTest. This increases the context that one needs to have to understand a test.

For the same reason, I wouldn't want other teams introducing semi-generic helper methods that apply some settings from their abstract classes. Whenever I find myself looking at random tests from other subprojects, I am always glad if they are "underengineered".

(especially in tests) Explicit is better than implicit